### ISSUE TITLE
 wrong input_signature: (TensorSpec(shape=(1, 1280, 1280, 3), dtype=tf.float32, name=None))

### ISSUE THREAD

**Issue by raphael-97** [ultralytics/yolov5#12037]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Export, PyTorch Hub

### Bug

Hello @glenn-jocher 

I have a problem with inference on an exported tensorflow modell with imgsz of 1280.

This is how i export my model: 
`
python export.py --weights best.pt --include saved_model --img 1280
`

Now i want to make inference on my image:

```
import torch

image_path = "test.png"

model = torch.hub.load("ultralytics/yolov5", "custom", "best_saved_model")

results = model(image_path)

```

same issue with 
`
results = model(image_path, size=1280)`

And i get this error message.
`
File "F:\\test.py", line 9, in <module>
   results = model(image_path)
File "F:\python310\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "F:\python310\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\.../.cache\torch\hub\ultralytics_yolov5_master\models\common.py", line 717, in forward
    y = self.model(x, augment=augment)  # forward
  File "F:\python310\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Users\.../.cache\torch\hub\ultralytics_yolov5_master\models\common.py", line 574, in forward
    y = self.model(im, training=False) if self.keras else self.model(im)
  File "F:\python310\lib\site-packages\tensorflow\python\saved_model\load.py", line 704, in _call_attribute
    return instance.__call__(*args, **kwargs)
  File "F:\python310\lib\site-packages\tensorflow\python\util\traceback_utils.py", line 153, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "F:\python310\lib\site-packages\tensorflow\python\eager\function_spec.py", line 531, in _convert_inputs_to_signature
    raise ValueError("Python inputs incompatible with input_signature:\n"
ValueError: Python inputs incompatible with input_signature:
  inputs: (
    [[[[    0.44706     0.44706     0.44706]
   [    0.44706     0.44706     0.44706]
   [    0.44706     0.44706     0.44706]
   ... 
input_signature: (
TensorSpec(shape=(1, 1280, 1280, 3), dtype=tf.float32, name=None)).
`

It looks like there is something wrong with the letterbox function, because in detect.py inference works and it creates an image with 1280x1280 shape. But not with torch.hub.load...

After some debugging i found out that the dataloaders.py script creates an input with shape 1280x1280 which works if i run detect.py
![bug1](https://github.com/ultralytics/yolov5/assets/63296115/58d1d6a7-99e1-4000-94ca-dd4d5a132a9c)

but in common.py if i do inference with the torch.hub version it does not create an input of 1280x1280:

![bug2](https://github.com/ultralytics/yolov5/assets/63296115/fdcbd4c3-b112-40cd-abe1-8e2d85d98ae5)

Something is definitely wrong here. Can you please fix it, i need it :D

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by VYRION-Ai** [comment]
What image size do you use in training?

**Comment by raphael-97** [comment]
@VYRION-Ai it's trained on 1280. It has nothing to do with training here. It's about a difference between torch.hub and detect.py to do inference with the model.
![Bug3](https://github.com/ultralytics/yolov5/assets/63296115/57d8309a-0eb1-4074-a27c-e8300974b819)

The exported model needs an input of 1280x1280. This works with detect.py but not with the inference in torch.hub.load and it should work as it is here: https://docs.ultralytics.com/yolov5/tutorials/model_export/#exported-model-usage-examples

In other words in the torch.hub variant the model gets the wrong input size and that's the reason for this error message...

**Comment by glenn-jocher** [comment]
@Glatzen95 Thank you for bringing this to our attention. The issue seems to be with the difference between using `torch.hub.load` and `detect.py` for inference with the exported model. 

The exported model requires an input size of 1280x1280, which works correctly with `detect.py` but not with `torch.hub.load`. This discrepancy can be seen in the provided images.

We apologize for any inconvenience caused. We will investigate this issue further and work towards resolving it. Your feedback is valuable in helping us improve the functionality of YOLOv5.

If you have any additional information or suggestions, please feel free to share. Thank you for your patience and understanding.

Best regards,

Glenn Jocher
Ultralytics YOLOv5 Repo Maintainer

**Comment by raphael-97** [comment]
@glenn-jocher No apologies needed. It would be nice if this gets fixed relatively quickly, because I need this for my thesis :) And keep up your great work!

**Comment by glenn-jocher** [comment]
@VYRION-Ai no problem at all! We understand the urgency, especially when it comes to important projects like your thesis. We will prioritize investigating and resolving this issue as quickly as possible. Thank you for your kind words and support, we truly appreciate it!

Please let us know if you have any further questions or if there's anything else we can assist you with. We're here to help!

Glenn Jocher - Ultralytics YOLOv5 Repo Maintainer

**Comment by raphael-97** [comment]
@glenn-jocher 

Hello. 

I also exported the model to TensorRT and the same problem seems to happen there with the torch.hub variant.


```somePath/best.engine for TensorRT inference...
[08/31/2023-13:09:14] [TRT] [I] Loaded engine size: 50 MiB
[08/31/2023-13:09:14] [TRT] [I] [MemUsageChange] TensorRT-managed allocation in engine deserialization: CPU +0, GPU +48, now: CPU 0, GPU 48 (MiB)
[08/31/2023-13:09:14] [TRT] [I] [MemUsageChange] TensorRT-managed allocation in IExecutionContext creation: CPU +0, GPU +136, now: CPU 0, GPU 184 (MiB)

  File "/home/local/.local/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/local/.local/lib/python3.8/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/local/.cache/torch/hub/ultralytics_yolov5_master/models/common.py", line 717, in forward
    y = self.model(x, augment=augment)  # forward
  File "/home/local/.local/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/local/.cache/torch/hub/ultralytics_yolov5_master/models/common.py", line 549, in forward
    assert im.shape == s, f"input size {im.shape} {'>' if self.dynamic else 'not equal to'} max model size {s}"
AssertionError: input size torch.Size([1, 3, 736, 1280]) not equal to max model size (1, 3, 1280, 1280)
```


When can we expect a fix, because i think no export works at the moment for torch.hub variant. 
Would be nice to get an update :)

**Comment by glenn-jocher** [comment]
@Glatzen95 hello,

Thank you for bringing this issue to our attention. We apologize for any inconvenience caused. We are aware of this problem and are actively working on resolving it. We understand that it is affecting all exports using the torch.hub variant at the moment.

We appreciate your patience and understanding. We will do our best to provide an update as soon as the fix is available.

If you have any further questions or concerns, please feel free to let us know.

Glenn Jocher - Ultralytics YOLOv5 Repo Maintainer
